src/components/metadata/AddPropertyButton.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  Button,
+  FormGroup,
+  HelperText,
+  HelperTextItem,
+  Popover,
+  Radio,
+  Split,
+  SplitItem,
+  Stack,
+  StackItem,
+  TextInput,
+  Title,
+  Tooltip,
+} from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { useState } from 'react';
+
+type AddPropertyPopoverProps = {
+  textLabel?: string;
+  model: any;
+  path: string[];
+  disabled?: boolean;
+  onChangeModel: (model: any) => void;
+};
+
+/**
+ * Add property button shows a popover to receive user inputs for new property name and type,
+ * as well as to validate if the property name already exists before actually adding it
+ * into the model object.
+ * @param props
+ * @constructor
+ */
+export function AddPropertyButton({
```

Comment:
Not really, just aligned to the same one allover the files - is there any rule for the usage?

Reply:
No, I don't think that we have a `linting` rule for that, it will be fine as long as we don't use the `this` keyboard since it will be pointing to the function and not to the outside scope, but I haven't seen that usage on this pull request :smiley:

Reply:
oh thanks good to know!